org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java

```diff
@@ -390,8 +384,6 @@ public ICompilationUnit handleOpen(DidOpenTextDocumentParams params) {
 							OpenableElementInfo elementInfo = (OpenableElementInfo) pkg.getElementInfo();
 							elementInfo.addChild(unit);
 						}
-					} else { // File not exists
-						return unit;
```

Comment:
This is the interesting bit (fix)

Reply:
Any reason to remove this?

Reply:
it's removed because we still want to trigger `triggerValidation(unit);` and others for units even if there is no IResource underlying. JDT can still validate such units fine.